x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/RestEqlSearchAction.java

```diff
@@ -46,6 +46,14 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             eqlRequest = EqlSearchRequest.fromXContent(parser);
             eqlRequest.indices(Strings.splitStringByCommaToArray(request.param("index")));
             eqlRequest.indicesOptions(IndicesOptions.fromRequest(request, eqlRequest.indicesOptions()));
+            if (request.hasParam("wait_for_completion_timeout")) {
+                eqlRequest.waitForCompletionTimeout(
+                    request.paramAsTime("wait_for_completion_timeout", eqlRequest.waitForCompletionTimeout()));
+            }
+            if (request.hasParam("keep_alive")) {
+                eqlRequest.keepAlive(request.paramAsTime("keep_alive", eqlRequest.keepAlive()));
+            }
+            eqlRequest.keepOnCompletion(request.paramAsBoolean("keep_on_completion", eqlRequest.keepOnCompletion()));
         }
 
         return channel -> client.execute(EqlSearchAction.INSTANCE, eqlRequest, new RestResponseListener<>(channel) {
```

Comment:
You should use `RestCancellableNodeClient` in order to get the automatic cancellation if the user closes the rest channel ?

Reply:
Good catch. I am going to open PR against master for that in a bit since it's not async execution-specific.

Reply:
Hmm, I need AbstractEqlBlockingIntegTestCase to do IT testing for it and it only exists in this branch for now. So, I think I will open a separate PR but against this branch.